app/src/main/java/org/greenstand/android/TreeTracker/BaseViewModel.kt

```diff
@@ -0,0 +1,6 @@
+package org.greenstand.android.TreeTracker
+
+import androidx.lifecycle.ViewModel
+import org.koin.core.KoinComponent
+
+abstract class BaseViewModel : ViewModel(), KoinComponent
```

Comment:
I did `KoinComponent` here because it'll make it easier for us to inject stuff into our viewModels whenever necessary. While I personally prefer if we can have all of it provided in Koin and then generated in the viewModelFactory so that we get that compile time speed boost, having the KoinComponent will give the implementer extra flexibility when building ViewModels for future features.

Reply:
Since that use case hasn't come up, I'd prefer if we didn't overdesign this. As it is now, all our viewmodels use constructor injection which doesn't need this.

Reply:
Works for me. Thanks for that call out! I sometimes need that reminder to dial it back. I can easily remove and keep it the way it is.

Reply:
Removed